sigstore/_cli.py

```diff
@@ -62,6 +62,19 @@ def __repr__(self) -> str:
         return f"{self._name} (embedded)"
 
 
+def _boolify_env(envvar: str) -> bool:
+    """
+    An `argparse` helper for turning an environment variable into a boolean.
+
+    The only valid `True` states are (case-insensitive) "true", "yes", and "1".
+    Anything else, including the absence of the variable, is `False`.
+    """
+    val = os.getenv(envvar)
+    if val is None:
+        return False
+    return val.lower() in {"true", "yes", "1"}
```

Comment:
Can we make this an implementation of https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool?

Reply:
Yep, doing that now.

Reply:
Done; the only deviation is for `None`, since we need that to default to `False` to prevent the absence of envvars from causing errors.

Reply:
N.B.: This doesn't produce the prettiest user errors, since the `ValueError` isn't caught in the normal `argparse` handler:

```console
$ SIGSTORE_OVERWRITE=x sigstore sign README.md
Traceback (most recent call last):
  File "/Users/william/devel/sigstore-python/env/bin/sigstore", line 8, in <module>
    sys.exit(main())
  File "/Users/william/devel/sigstore-python/sigstore/_cli.py", line 284, in main
    parser = _parser()
  File "/Users/william/devel/sigstore-python/sigstore/_cli.py", line 190, in _parser
    default=_boolify_env("SIGSTORE_OVERWRITE"),
  File "/Users/william/devel/sigstore-python/sigstore/_cli.py", line 83, in _boolify_env
    raise ValueError(f"can't coerce '{val}' to a boolean")
ValueError: can't coerce 'x' to a boolean
```